docs/references/api/aggregate_functions.rst

```diff
@@ -0,0 +1,228 @@
+
+Using Embedded Resources as Grouping Columns
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Using an embedded resource as a grouping column allows you to use data from an association to group the results of an aggregation.
+
+For example, imagine that the ``orders`` table from the examples above is related to a ``customers`` table. If you want to get the sum of the ``amount`` column grouped by the ``name`` column from the ``customers`` table, you can include the customer name, using the standard :ref:`resource_embedding` syntax, and perform a sum on the ``amount`` column.
+
+.. tabs::
+
+  .. code-tab:: http
+
+    GET /orders?select=amount.sum(),customers(name) HTTP/1.1
+
+  .. code-tab:: bash Curl
+
+    curl "http://localhost:3000/orders?select=amount.sum(),customers(name)"
+
+You will then get the summed amount, along with the embedded customer resource:
+
+.. code-block:: json
+
+  [
+    {"sum": 100.0, "customers": {"name": "Customer A"}},
+    {"sum": 200.0, "customers": {"name": "Customer B"}}
+  ]
+
+.. note::
+ The previous example uses a has-one association to demonstrate this functionality, but you may also use has-many associations as grouping columns, although there are few obvious use cases for this.
+
+Using Aggregate Functions Within the Context of an Embedded Resource
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When embedding a resource, you can apply aggregate functions to columns from the associated resource to perform aggregations within the context of an embedded resource.
+
+Continuing with the example relationship between ``orders`` and ``customers`` from the previous section, imagine that you want to fetch the ``name``, ``city``, and ``state`` of all customers, along with the sum of their daily order amounts. This can be done in the following way:
+
+.. tabs::
+
+  .. code-tab:: http
+
+    GET /customers?select=name,city,state,orders(amount.sum(),order_date) HTTP/1.1
+
+  .. code-tab:: bash Curl
+
+    curl "http://localhost:3000/customers?select=name,city,state,orders(amount.sum(),order_date)"
+
+TODO: Add results.
+
+In this example, the ``amount`` column is summed and grouped by the ``order_date`` *within* the context of the embedded resource. That is, the ``name``, ``city``, and ``state`` from the ``customers`` table have no bearing on the aggregation performed in the context of the ``orders`` association; instead, the ``orders`` table is implicitly grouped by customer.
+
+Using Columns from a Spread Resource
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When you :ref:`spread an embedded resource <spread_embed>`, the columns from the embedded resource that has been spread are treated as if they were columns of the top-level resource, both when using them as grouping columns and when applying aggregate functions to them.
```

Comment:
@steve-chavez what is the proper terminology here for: "the embedded resource that has been spread"? would it be "the spreaded resource" or something else? finding it a bit awkward in this section

Reply:
thanks!